simplex/comm.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/ava-labs/simplex"
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/message"
+	"github.com/ava-labs/avalanchego/proto/pb/p2p"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/snow/networking/sender"
+	"github.com/ava-labs/avalanchego/subnets"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var (
+	_               simplex.Communication = (*Comm)(nil)
+	errNodeNotFound                       = errors.New("node not found in the validator list")
+)
+
+type Comm struct {
+	logger   simplex.Logger
+	subnetID ids.ID
+	chainID  ids.ID
+	// nodeID is this nodes ID
+	nodeID simplex.NodeID
+	// nodes are the IDs of all the nodes in the subnet
+	nodes []simplex.NodeID
+	// sender is used to send messages to other nodes
+	sender     sender.ExternalSender
+	msgBuilder message.OutboundMsgBuilder
+}
+
+func NewComm(config *Config) (*Comm, error) {
+	nodes := make([]simplex.NodeID, 0, len(config.Validators))
+
+	// grab all the nodes that are validators for the subnet
+	for _, vd := range config.Validators {
+		nodes = append(nodes, vd.NodeID[:])
+	}
+
+	if _, ok := config.Validators[config.Ctx.NodeID]; !ok {
+		config.Log.Warn("Node is not a validator for the subnet",
+			zap.String("nodeID", config.Ctx.NodeID.String()),
+			zap.String("chainID", config.Ctx.ChainID.String()),
+			zap.String("subnetID", config.Ctx.SubnetID.String()),
+		)
+		return nil, fmt.Errorf("%w could not find our node: %s", errNodeNotFound, config.Ctx.NodeID)
```

Comment:
"node not found in the validator list could not find our node: NodeID-xxx"

It reads a bit weirdly to me. Perhaps:
```suggestion
		return nil, fmt.Errorf("our %w: %s", errNodeNotFound, config.Ctx.NodeID)
```

"our node not found in the validator list: NodeID-xxx"